src/use_promise_guard/index.es

```diff
@@ -0,0 +1,19 @@
+import useMounted from '../use_mounted';
+
+const usePromiseGuard = (deps) => {
+  const isMounted = useMounted(deps);
+
+  const execute = (getPromise) => new Promise((resolve, reject) => {
+    getPromise(resolve, reject)
+      .then((result) => {
+        if (isMounted.current) resolve(result);
```

Comment:
if unmounted promise stays unresolved. Wouldn't it cause a memory leak?

Reply:
it's just sugar over callbacks, so should be ok as long as there are no listeners etc

Reply:
https://alexn.org/blog/2017/10/11/javascript-promise-leaks-memory.html